File: Unidades/Unidad2/palindrome.cpp
#include <iostream>
#include <string>

using namespace std;

bool isPalindrome(string str) {
    if (str.length() <= 1) {
        return true;
    }
    if (str[0] == str[str.length()-1]) {
        return isPalindrome(str.substr(1, str.length()-2));
    }
    return false;
}

int main() {
    string input;
    cout << "Enter a word or phrase: ";
    getline(cin, input);
    if (isPalindrome(input)) {
        cout << input << " is a palindrome" << endl;
    } else {
        cout << input << " is not a palindrome" << endl;
    }
    return 0;
}

File: Unidades/Unidad2/palindrome.py
def is_palindrome(s):
    if len(s) <= 1:
        return True
    if s[0] == s[-1]:
        return is_palindrome(s[1:-1])
    return False


input_str = input("Enter a word or phrase: ")
if is_palindrome(input_str):
    print(input_str, "is a palindrome")
else:
    print(input_str, "is not a palindrome")



